scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,567 @@
+/* https://build.libelektra.org/jenkins/job/elektra-jenkinsfile/
+ * This file describes how the elektra-jenkinsfile buildjob should be
+ * executed.
+ *
+ * 1. imports and global variables are set
+ * 2. define the main stages of the pipeline
+ * 3. describe sub stages. This is where you will want to add new builds
+ * 4. helper section to help write build scripts
+ */
+
+// FIXME sloc publish errors
+// XXX add missing jobs
+// TODO have a per plugin/binding deps in Dockerfile for easier maintanence
+// TODO add warnings plugins to scan for compiler warnings
+
+// Imports
+import java.text.SimpleDateFormat
+
+// Buildjob properties
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '60', artifactNumToKeepStr: '60'))
+])
+
+// Globals
+docker_node_label = 'docker'
+docker_image_prefix = 'build-elektra-'
+
+DOCKER_IMAGES = [
+    sid: [
+        id: docker_image_prefix + 'debian-sid',
+        context: "./scripts/docker/debian/sid",
+        file: "./scripts/docker/debian/sid/Dockerfile"
+    ],
+    stretch: [
+        id: docker_image_prefix + 'debian-stretch',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile",
+    ],
+    stretch_minimal: [
+        id: docker_image_prefix + 'debian-stretch-minimal',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.minimal",
+    ],
+    stretch_doc: [
+        id: docker_image_prefix + 'debian-stretch-doc',
+        context: "./scripts/docker/debian/stretch",
+        file: "./scripts/docker/debian/stretch/Dockerfile.doc",
+    ],
+    jessie: [
+        id: docker_image_prefix + 'debian-jessie',
+        context: "./scripts/docker/debian/jessie",
+        file: "./scripts/docker/debian/jessie/Dockerfile",
+    ],
+    xenial: [
+        id: docker_image_prefix + 'ubuntu-xenial',
+        context: "./scripts/docker/ubuntu/xenial",
+        file: "./scripts/docker/ubuntu/xenial/Dockerfile",
+    ]
+]
+
+// Define reusable cmake Flag globals
+CMAKE_FLAGS_BASE = [
+    'SITE': '${STAGE_NAME}',
+    'KDB_DB_SYSTEM': '${HOME}/.config/kdb/system',
+    'KDB_DB_SPEC': '${HOME}/.config/kdb/spec',
+    'CMAKE_INSTALL_PREFIX': '${WORKSPACE}/system',
+    'INSTALL_SYSTEM_FILES': 'OFF',
+    'BUILD_DOCUMENTATION': 'OFF'
+]
+
+CMAKE_FLAGS_BUILD_ALL = [
+    'BINDINGS': 'ALL;EXPERIMENTAL;-DEPRECATED',
```

Comment:
EXPERIMENTAL is automatically added if not removed. Why calling it ALL if DEPRECATED is removed?

Reply:
> EXPERIMENTAL is automatically added if not removed.

Pretty sure it wasn't including something hence it is stated explicitly. Deprecated was removed because gi threw test errors.

Reply:
> Pretty sure it wasn't including something hence it is stated explicitly.

That would be a bug, please report it.

> Deprecated was removed because gi threw test errors.

Isn't not installing gobject-introspection the easier way? Or do you have a check if all plugins were actually added? (Would be a useful check...)

Reply:
> EXPERIMENTAL is automatically added if not removed.

https://build.libelektra.org/jenkins/blue/organizations/jenkins/elektra-jenkinsfile/detail/elektra-jenkinsfile/187/pipeline/150

Reply:
ah now i remember, it was because of the haskell plugin and has been reported in #1890 and an open pr exists at #1894

Reply:
Ahh, perfect. Then simply add a TODO that this needs to be removed after  #1894 is merged :smile: